### PR TITLE
feat: Add SplashScreen plugin

### PR DESCRIPTION
**Pull request by jcesarmobile**
closes https://github.com/ionic-team/capacitor-plugins/issues/133
closes https://github.com/ionic-team/capacitor-plugins/issues/150

### CODE REVIEW

File: splash-screen/src/web.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import type {
+  SplashScreenPlugin,
+  SplashScreenShowOptions,
+  SplashScreenHideOptions,
+} from './definitions';
+
+export class SplashScreenWeb extends WebPlugin implements SplashScreenPlugin {
+  async show(_options?: SplashScreenShowOptions): Promise<void> {
+    return undefined;
```
Comment (dwieeb) [inline splash-screen/src/web.ts:11]:
```suggestion
    throw this.unimplemented('Not implemented for web.');
```

File: splash-screen/src/web.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  }
+
+  async hide(_options?: SplashScreenHideOptions): Promise<void> {
+    return undefined;
```
Comment (dwieeb) [inline splash-screen/src/web.ts:15]:
```suggestion
    throw this.unimplemented('Not implemented for web.');
```

File: splash-screen/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Contributing
```
Comment (dwieeb) [inline splash-screen/CONTRIBUTING.md:1]: This file can be removed

File: splash-screen/src/web.ts (hunk 1 above)
Comment (jcesarmobile) [reply splash-screen/src/web.ts:11]: it was like that, so I didn't want to change the behavior

File: splash-screen/android/src/main/java/com/capacitorjs/plugins/splashscreen/SplashScreen.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+import com.getcapacitor.Logger;
+
+/**
+ * A Splash Screen service for showing and hiding a splash screen in the app.
+ */
+public class SplashScreen {
+
+    private ImageView splashImage;
+    private ProgressBar spinnerBar;
+    private WindowManager wm;
```
Comment (carlpoole) [inline splash-screen/android/src/main/java/com/capacitorjs/plugins/splashscreen/SplashScreen.java:28]: rename to windowManager

**Comment by carlpoole** [comment]
Couple of ideas

`SplashScreenConfig` and `SplashScreenSettings` could use Builders so the resulting object is immutable - it looks like those are only accessed inside SplashScreen and not set there.

the `show` method in SplashScreen might also benefit from a method signature without a Settings parameter so it just uses a default SplashScreenSettings. (similar to the old one that just took an activity like this:
 `public static void show(final Activity a) {		
         show(a, DEFAULT_LAUNCH_SHOW_DURATION, DEFAULT_FADE_IN_DURATION, DEFAULT_FADE_OUT_DURATION, DEFAULT_AUTO_HIDE, null, null);		
     }`)

**Comment by jcesarmobile** [comment]
As the idea of the implementation class is to make it easier to reuse the plugin by native code, it's easier to have a single object rather than a bunch of params.
Also, iOS had a TODO about the show having too many parameters, which wasn't good for the linter and the TODO comment suggested to use a struct, so I did. So on Android I used classes since there are no structs.

My code is based on Ian's approach on camera PR, which was approved by you and Dan, so if we want to use builders his PR should also be changed.

I removed the show with no params because it was not being used and iOS didn't have an equivalent, but I can put it back if you want. But then iOS should also have something like that for consistency.

**Comment by carlpoole** [comment]
> I removed the show with no params because it was not being used and iOS didn't have an equivalent, but I can put it back if you want. But then iOS should also have something like that for consistency.

I'm thinking from a reusability standpoint for this method:

```show(final AppCompatActivity activity, final SplashScreenSettings settings, final SplashListener splashListener)```

You would call it:

`show(activity, new SplashScreenSettings(), listener)`

or you could also do

`show(activity, listener)` 

if you just want defaults. If the first one is acceptable then that's fine.

**Review by dwieeb — APPROVED** [review]

**Pull request merged**